app/ante/evm/claim_rewards_test.go

```diff
@@ -0,0 +1,149 @@
+				)
+				suite.Require().NoError(err, "failed to fund distribution module account")
+				suite.app.AccountKeeper.SetModuleAccount(suite.ctx, distrAcc)
+
+				// end block and increase block height
+				staking.EndBlocker(suite.ctx, suite.app.StakingKeeper)
+				suite.ctx = suite.ctx.WithBlockHeight(suite.ctx.BlockHeight() + 1)
+
+				// allocate rewards to validator
+				validator := suite.app.StakingKeeper.Validator(suite.ctx, valAddr)
+				suite.app.DistrKeeper.AllocateTokensToValidator(suite.ctx, validator, sdk.NewDecCoins(sdk.NewDecCoin(utils.BaseDenom, sdk.NewInt(1000))))
+
+				// check that the historical count is 3 (initial creation, delegation + reward allocation)
+				historicalCount := suite.app.DistrKeeper.GetValidatorHistoricalReferenceCount(suite.ctx)
+				suite.Require().Equal(uint64(3), historicalCount, "historical count should be 3; got %d", historicalCount)
+			},
+			amount:      100,
+			expErr:      false,
+			errContains: "",
+		},
+		{
+			name:        "fail - no staking rewards to claim",
+			malleate:    func(valAddr sdk.ValAddress) {},
+			amount:      100,
+			expErr:      true,
+			errContains: "insufficient staking rewards to cover transaction fees",
+		},
+	}
+
+	// ----------------------------------------
+	// Run testcases
+	//
+	for _, tc := range testcases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			addr, valAddr := suite.BasicSetupForClaimRewardsTest()
+			tc.malleate(valAddr)
+			amount := sdk.NewCoins(sdk.NewCoin(utils.BaseDenom, sdk.NewInt(tc.amount)))
+
+			// NOTE: this means that the delegation rewards are only available for the val address itself
+			// seems like a self reward, that is done with the AllocateTokensToValidator function -> might need to use a different function
+			rewards, err := suite.app.DistrKeeper.WithdrawDelegationRewards(suite.ctx, sdk.AccAddress(valAddr), valAddr)
+			suite.Require().NoError(err, "failed to withdraw delegation rewards")
+			suite.Require().Equal(sdk.NewCoins(sdk.NewCoin(utils.BaseDenom, sdk.NewInt(1230))), rewards, "expected rewards to be withdrawn")
+
+			// TODO: remove logging
+			suite.T().Logf("delegations: %v", suite.app.StakingKeeper.GetAllDelegatorDelegations(suite.ctx, addr))
+			err = evm.ClaimSufficientStakingRewards(suite.ctx, suite.app.StakingKeeper, suite.app.DistrKeeper, addr, amount)
+
+			if tc.expErr {
+				suite.Require().Error(err)
+				suite.Require().ErrorContains(err, tc.errContains)
+			} else {
+				suite.Require().NoError(err)
+			}
+		})
+	}
+}
+
+// BasicSetupForClaimRewardsTest is a helper function, that creates a validator and a delegator
+// and funds them with some tokens. It also sets up the staking keeper to include a self-delegation
+// of the validator and a delegation from the delegator to the validator.
+func (suite *AnteTestSuite) BasicSetupForClaimRewardsTest() (sdk.AccAddress, sdk.ValAddress) {
```

Comment:
Let's avoid adding more setup functionality and use what we have available on the testutils